samples/core/pom.xml

```diff
@@ -213,6 +213,7 @@
     <dependency>
       <groupId>org.springframework.boot</groupId>
       <artifactId>spring-boot-starter-web</artifactId>
+      <scope>provided</scope>
```

Comment:
I am fine with this but I would have expected to remove this here and also move the app to the server module. But this will have other implications. So just some thoughts for discussion and no rejection of your change that is fine.